es6-classes/student.js

```diff
@@ -0,0 +1,29 @@
+/* exported Student */
+
+class Student {
+  constructor(firstName, lastName, subject) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.subject = subject;
+  }
+
+  getFullName(firstName, lastName) {
+    return `${firstName} ${lastName}`;
+  }
+
+  getIntroduction(getFullName, subject) {
+    return `Hello my name is ${getFullName} and I am studying ${subject}`;
+  }
+}
+
+Student.prototype.getFullName = function () {
+  const { firstName, lastName } = this;
+  return `${firstName} ${lastName}`;
+};
+
+Student.prototype.getIntroduction = function () {
+  const { subject } = this;
+  const fullName = this.getFullName();
+  return `Hello, my name is ${fullName} and I am studying ${subject}.`;
+
+};
```

Comment:
You can remove these, they should be defined in the class body.

Reply:
ok, fixed it